packages/frontend/src/components/CreateSwap/CreateSwap.tsx

```diff
@@ -164,6 +166,16 @@ const CreateSwap = () => {
   const depositMax = isConnected ? spendableBalance : undefined;
   const selectedFromTokenWithNetwork = getTokenWithNetwork(selectedFromToken, fromChain);
   const selectedToTokenWithNetwork = getTokenWithNetwork(selectedToToken, toChain);
+  const fromUsdValue = useUsdValue({
+    address: fromToken?.address,
+    chainId: fromChain.id,
+    amount: fromAmount,
+  });
+  const toUsdValue = useUsdValue({
+    address: toToken?.address,
```

Comment:
If this is just to be used for to and from tokens, we could be most of these values from existing contexts.
I think your approach here is better for reusability though.

Reply:
It might be used for more cases in the future, once Sifi becomes bigger, so I turned it into a reusable hook because of that. 👌🏻